components/branding-preference-mgt/org.wso2.carbon.identity.branding.preference.mgt/src/main/java/org/wso2/carbon/identity/branding/preference/mgt/BrandingPreferenceManagerImpl.java

```diff
@@ -0,0 +1,324 @@
+import static org.wso2.carbon.identity.branding.preference.mgt.util.BrandingPreferenceMgtConstants.ErrorMessages.ERROR_CODE_UNSUPPORTED_ENCODING_EXCEPTION;
+import static org.wso2.carbon.identity.branding.preference.mgt.util.BrandingPreferenceMgtConstants.RESOURCE_ALREADY_EXISTS_ERROR_CODE;
+import static org.wso2.carbon.identity.branding.preference.mgt.util.BrandingPreferenceMgtConstants.RESOURCE_NAME_SEPARATOR;
+import static org.wso2.carbon.identity.branding.preference.mgt.util.BrandingPreferenceMgtConstants.RESOURCE_NOT_EXISTS_ERROR_CODE;
+import static org.wso2.carbon.identity.branding.preference.mgt.util.BrandingPreferenceMgtUtils.handleClientException;
+import static org.wso2.carbon.identity.branding.preference.mgt.util.BrandingPreferenceMgtUtils.handleServerException;
+
+/**
+ * Branding Preference Management service implementation.
+ */
+public class BrandingPreferenceManagerImpl implements BrandingPreferenceManager {
+
+    private static final Log log = LogFactory.getLog(BrandingPreferenceManagerImpl.class);
+    //TODO: Improve API to manage application level & language level theming resources in addition to the tenant level.
+
+    @Override
+    public BrandingPreference addBrandingPreference(BrandingPreference brandingPreference)
+            throws BrandingPreferenceMgtException {
+
+        String resourceName = getResourceName
+                (brandingPreference.getType(), brandingPreference.getName(), brandingPreference.getLocale());
+        String tenantDomain = getTenantDomain();
+        // Check whether a branding resource already exists with the same name in the particular tenant to be added.
+        if (isResourceExists(BRANDING_RESOURCE_TYPE, resourceName)) {
+            throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_ALREADY_EXISTS, tenantDomain);
+        }
+        String preferencesJSON = generatePreferencesJSONFromPreference(brandingPreference.getPreference());
+        if (!BrandingPreferenceMgtUtils.isValidJSONString(preferencesJSON)) {
+            throw handleClientException(ERROR_CODE_INVALID_BRANDING_PREFERENCE, tenantDomain);
+        }
+
+        try {
+            InputStream inputStream = BrandingPreferenceMgtUtils.generatePreferenceInputStream(preferencesJSON);
+            Resource brandingPreferenceResource = buildResourceFromBrandingPreference(resourceName, inputStream);
+            getConfigurationManager().addResource(BRANDING_RESOURCE_TYPE, brandingPreferenceResource);
+        } catch (ConfigurationManagementException e) {
+            if (RESOURCE_ALREADY_EXISTS_ERROR_CODE.equals(e.getErrorCode())) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Branding preferences are already exists for tenant: " + tenantDomain, e);
+                }
+                throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS, tenantDomain);
```

Comment:
Is this error code correct? Error checked in the if condition does not match here.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/3870/commits/d8b98f955a8d30d3fb07768ec4b1e5503aa46a48